flytekit/core/data_persistence.py

```diff
@@ -101,6 +103,24 @@ def get_fsspec_storage_options(
     return {}
 
 
+@decorator
+def retry_request(func, *args, **kwargs):
+    # TODO: Remove this method once s3fs has a new release. https://github.com/fsspec/s3fs/pull/865
+    retries = kwargs.pop("retries", 5)
+    for retry in range(retries):
+        try:
+            if retry > 0:
+                sleep(min(random.random() + 2 ** (retry - 1), 32))
```

Comment:
exponential backoff + jitter is usually implemented like:
```
sleep(random.randint(0, min(2 ** retry, 32)))
```

https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/ is the gold standard describing a few techniques for implementing exp. backoff + jitter.